### PR TITLE
Resolve issues found while enabling image test cases in osbuild

### PR DESCRIPTION
**Pull request by thozza**
There have been two main issues with the generated test cases:

1. The `osbuild` used to generate image test cases in F34 repos is too old and does not contain the fix for the qemu stage and qcow compatibility version. Solve this situation by adding a new option `--repofrompath` to `generate-all-test-cases` allowing to specify additional DNF repositories, which will be used on the Runner when installing any packages (such as osbuild).
2. In some cases, e.g. when analysing an extracted `tar` image, it could happen that the tools used by `image-info` to analyse the image could modify its content (e.g. create new files which were originally not on the image). This is especially an issue with `rpm`, which DB backend seemed to create files in `/var/lib/rpm/` when run. Ensure that the analysed directory can not be modified by bind-mounting it as a read only, before the analysis.

Related to https://github.com/osbuild/osbuild/pull/810

- [x] Regenerate all relevant image test cases 

This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [x] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

**Comment by thozza** [comment]
Test case regeneration is currently blocked on unavailability of `osbuild` upstream builds for ppc64le (https://github.com/osbuild/osbuild/pull/838)

File: test/README.md [hunk 1]
```diff
@@ -101,16 +101,32 @@ In simplified example, the script does the following:
...
+`--arch`, `--distro` or `--image-type` options. These only filter the default
+matrix. So to e.g. generate all image test cases for RHEL-8.5, simply run:
+
+```bash
+$ ./tools/test-case-generators/generate-all-test-cases \
+        --output test/data/manifests \
+        --distro rhel-85 \
+        <COMMAND> \
+        ...
+```
```
Comment (achilleas-k) [inline test/README.md:121]: Thanks for this :smile:

File: tools/test-case-generators/generate-all-test-cases [hunk 2]
```diff
@@ -1289,6 +1317,14 @@ def get_args():
             help="Don't delete the workdir created on the remote host after finishing.",
             default=False
         )
+    parser.add_argument(
+            "--repofrompath",
```
Comment (achilleas-k) [inline tools/test-case-generators/generate-all-test-cases:1321]: Why `repofrompath` and not something like `rpmrepo` or `extrarepo`?  I feel like the `frompath` part implies that the value should be a filesystem path.

**Review by achilleas-k — APPROVED** [review]
Looks good :+1: 
Though I do have one question about the new clarg.

File: tools/test-case-generators/generate-all-test-cases (hunk 2 above)
Comment (thozza) [reply tools/test-case-generators/generate-all-test-cases:1321]: From `man dnf`:
```
 --repofrompath <repo>,<path/url>
              Specify a repository to add to the repositories for this query.  This option can be used multiple times.
```

**Review by ondrejbudai — APPROVED** [review]

**Pull request merged**